### PR TITLE
feat: add cronjob to circle ci pipeline [TAROT-2376]

### PR DESCRIPTION
**Pull request by Afonsobli**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -35,3 +82,37 @@ workflows:
...
+    jobs:
+      - codacy/checkout_and_version
+      - build_and_publish_docker:
+          name: publish_docker_local
+          requires:
+            - codacy/checkout_and_version
+      - install_trivy_and_download_dbs:
+          name: install_trivy_and_download_dbs
+          requires:
+            - publish_docker_local
```
Comment (afsmeira) [inline .circleci/config.yml:104]:
```suggestion
      - install_trivy_and_download_dbs:
          name: install_trivy_and_download_dbs
          requires:
            - codacy/checkout_and_version
      - build_and_publish_docker:
          name: publish_docker_local
          requires:
            - install_trivy_and_download_dbs
      
```

It needs to be the other way around. First download the images to a well known directory, then copy the contents of the directory to the Docker image when building

File: .circleci/config.yml [hunk 2]
```diff
@@ -4,17 +4,64 @@ orbs:
...
+    docker:
+      # https://hub.docker.com/r/cimg/base this image is the base image on which the docker commands will run
+      - image: cimg/base:2023.04
+    working_directory: ~/workdir
+    steps:
+      - attach_run_command_and_persist:
+          cmd_name: Install trivy and download DBs
+          # The Java DB is cached for 3 days. Using the '--reset' flag clears the DB cache.
+          cmd: |
+            curl -sfL https://raw.githubusercontent.com/aquasecurity/trivy/main/contrib/install.sh | sh -s -- -b . v0.44.1
```
Comment (afsmeira) [inline .circleci/config.yml:54]:
```suggestion
            # https://aquasecurity.github.io/trivy/v0.45/getting-started/installation/#install-script
            curl -sfL https://raw.githubusercontent.com/aquasecurity/trivy/main/contrib/install.sh | sh -s -- -b . v0.44.1
```

Comment (Afonsobli) [reply .circleci/config.yml:54]: thank you!

**Review by afsmeira — APPROVED** [review]
LGTM. Did the scheduled workflow already run?

**Review by afsmeira — CHANGES_REQUESTED** [review]
Wait, you need to change the Dockerfile to copy the cache folder inside the image.

**Comment by Afonsobli** [comment]
> LGTM. Did the scheduled workflow already run?

it did!

File: Dockerfile [hunk 3]
```diff
@@ -10,6 +10,7 @@ RUN go build -o bin/codacy-trivy -ldflags="-s -w" ./cmd/tool
 RUN go run ./cmd/docgen
 
 COPY docs/ /docs/
+COPY /cache /dist/cache/codacy-trivy
```
Comment (afsmeira) [inline Dockerfile:13]:
```suggestion
COPY cache/ /dist/cache/codacy-trivy
```

**Review by afsmeira — COMMENTED** [review]
I think you need to add a `cache` entry in `.gitignore`.

File: Dockerfile [hunk 4]
```diff
@@ -9,15 +9,13 @@ RUN go mod verify
...
-RUN chown -R docker:docker /docs
-
-FROM busybox
 
 COPY --from=builder /src/bin /dist/bin
-COPY --from=builder /docs /docs
-COPY --from=builder /etc/passwd /etc/passwd
+COPY --from=builder /src/docs /docs 
+COPY --chown=docker:docker /docs /docs/
+COPY --from=builder /src/cache/ /dist/cache/codacy-trivy
```
Comment (afsmeira) [inline Dockerfile:19]:
```
COPY --from=builder --chown=docker:docker /src/bin /dist/bin
COPY --from=builder --chown=docker:docker /src/docs /docs 
COPY --from=builder --chown=docker:docker /src/cache/ /dist/cache/codacy-trivy
```

**Review by afsmeira — APPROVED** [review]

**Review by afsmeira — APPROVED** [review]

**Pull request merged**